### PR TITLE
VCM-7568 Enabling polygon tool for AOI selection

### PR DESCRIPTION
**Pull request by msw-ursaspace**
**What's Changed**

- Enabled polygon selection in the Analytic Ordering Toolbox
-- Tool tips for Area Of Interest and Buffer (km) have been updated
--- AOI Tool-tip now reads: "Point or Polygon Geometries Only"
--- Buffer (km) Tool-tip now reads: "The required image coverage area radius around the AOI point provided, in kilometers (km). Value must be between 1 and 5. The value is not used on polygon AOIs.
- Enabled polygon selection in the Tasking Toolbox
-- Tool tips for Area Of Interest and Buffer (km) have been updated
- Added new unit test to ensure the geojson helper properly processed a polygon shape file
- Updated .gitignore to ignore Visual Studio metadata directory
---
**To Test**

- Run all unit test (see readme)
- Submit an analytic order using a polygon AOI

Steps to Submit an analytic order using a polygon AOI:

1. In the Analytics toolbox select the "Create Feature" button to the right of the Area of Interest field (the pencil icon) and select "Polygons" (see first image below)
2. Add the polygon to the Map layer (seem second image below) this will add a "Analytics Area of Interest (Polygons) feature layer to the Contents pane (see third image below)
3. In the Analytics toolbox select the "Analytics Area of Interest (Polygons)" feature layer in the AOI dropdown
4. Complete the rest of the Analytics order form and hit run

Similar steps can be taken to submit a tasking order.

image
Image 1: Analytics Toolbox

image
Image 2: Map Layer

image
Image 3: Contents Pane

image
Image 4: AOI Tool-tip Text

image
Image 5: Buffer (km) Tool-tip Text

### CODE REVIEW

**Comment by ursaspace** [comment]
Screenshot 2023-09-01 at 9 26 04 AM

I know we talked about this tooltip already, but I would change this first sentence to `The required image coverage area radius around a point AOI` and I'd take out the "provided", or add a "when provided".

I think it's more clear when we list point before AOI so that people actually read that we're specifying points here and then talking about polygons later.

**Comment by ursaspace** [comment]
This is out-of-scope, so this would be a follow up ticket, but it might be nice to make the jira ticket description more readable/make it more clear that there's a list of multiple features. Right now we just display them like:

`AOIs:
POLYGON ((-76.49062552408395 42.44635670415125, -76.518016023971 42.443614745504206, -76.5153619065483 42.42810083308185, -76.4875467469753 42.42935462482989, -76.49062552408395 42.44635670415125))
POLYGON ((-76.52565988484332 42.46116120811279, -76.51950233062603 42.44706175966407, -76.49264265422352 42.457714969204055, -76.49975569071297 42.466173580059966, -76.52565988484332 42.46116120811279))
`
but it might be more clear if we indicated this was a list with the [] and comma separated.

**Comment by msw-ursaspace** [comment]
Updated Analytic and Tasking toolbox buffer field tooltip per feedback:
image

File: src/parameters/aoi.py [hunk 1]
```diff
@@ -11,7 +11,7 @@ def aoi_parameter():
         parameterType="Required",
         direction="Input",
     )
-    param.filter.list = ["Point"]
+    param.filter.list = ["Point", "Polygon"]
```
Comment (ursaspace) [inline src/parameters/aoi.py:14]: We used to not accept polygon for new tasking orders, is that still true? Maybe we make two separate aoi parameters in that case, one that contains polygons and can be used in the analytic ordering tool and then one with just points that's used in the tasking tool

Comment (ursaspace) [reply src/parameters/aoi.py:14]: Resolving this based of conversation offline. We do now accept polygons in tasking so this should all be happy :)

**Review by ursaspace — APPROVED** [review]

**Pull request merged**